src/ParameterBag.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace SymfonyDocsBuilder;
+
+/**
+ * ParameterBag is a container for key/value pairs.
+ */
+class ParameterBag implements \IteratorAggregate, \Countable
+{
+    /**
+     * Parameter storage.
+     */
+    protected $parameters;
+
+    /** @var bool */
+    private $initialized = false;
```

Comment:
Needed? It looks like everything is required (and it should be). I do see a call to initialize() in a different class - I don’t like that - if this is a service that holds some variables, it should be fully populated immediately by the container. Also... if we can think of a better name that describes what it holds in our app, I’d like that too :)

Reply:
the point here is that we can't populate it immediately by the container, because the parameters it holds are the ones we're passing to the command as parameter (sourceDir, outputDir...)
these are parsed once the command is started (and the container's parameterBag is frozen...)